modules/phase_field/src/action/ConservedAction.C

```diff
@@ -0,0 +1,175 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ConservedAction.h"
+
+template <>
+InputParameters
+validParams<ConservedAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription(
+      "Set up the variable(s) and the kernels needed for a conserved phase field variable."
+      " Note that for a direct solve, the element family and order are overwritten with hermite "
+      "and third.");
+  MooseEnum solves("direct split");
+  params.addRequiredParam<MooseEnum>("solve_type", solves, "Split or direct solve?");
+  params.addRequiredParam<NonlinearVariableName>("variable_name",
+                                                 "The name of the nonconserved variable");
+  MooseEnum families("LAGRANGE MONOMIAL HERMITE HIERARCHIC CLOUGH XYZ SZABAB"
+                     "BERNSTEIN L2_LAGRANGE L2_HIERARCHIC",
+                     "LAGRANGE");
+  params.addParam<MooseEnum>("family",
+                             families,
+                             "Specifies the family of FE "
+                             "shape functions to use for this variable");
+  MooseEnum orders("CONSTANT FIRST SECOND THIRD FOURTH", "FIRST");
+  params.addParam<MooseEnum>("order",
+                             orders,
+                             "Specifies the order of the FE "
+                             "shape function to use for this variable");
+  params.addParam<Real>("scaling", 1.0, "Specifies a scaling factor to apply to this variable");
+  params.addParam<bool>("implicit", true, "Whether kernels are implicit or not");
+  params.addParam<bool>(
+      "use_displaced_mesh", false, "Whether to use displaced mesh in the kernels");
+  params.addParamNamesToGroup("scaling implicit use_displaced_mesh", "Advanced");
+  params.addRequiredParam<MaterialPropertyName>("mobility", "The mobility used with the kernel");
+  params.addParam<std::vector<VariableName>>("args",
+                                             "Vector of variable arguments this kernel depends on");
+  params.addRequiredParam<MaterialPropertyName>(
+      "free_energy", "Base name of the free energy function F defined in a free energy material");
+  params.addRequiredParam<MaterialPropertyName>("kappa", "The kappa used with the kernel");
+
+  return params;
+}
+
+ConservedAction::ConservedAction(const InputParameters & params)
+  : Action(params),
+    _solve_type(getParam<MooseEnum>("solve_type")),
+    _var_name(getParam<NonlinearVariableName>("variable_name")),
+    _scaling(getParam<Real>("scaling"))
+{
+  if (_solve_type == 0) // Direct solve must be third order hermite
```

Comment:
It is pretty convenient to have the element types set automatically for a direct solve because you then only have to change solve_type from split to direct in this block. We could throw an error if the user provides a family and order that aren't third and hermite, but I'm not sure how to see if the user provides values for those since we do want to have default values.